core-library/src/main/java/org/silverpeas/core/contribution/content/form/record/GenericRecordSetManager.java

```diff
@@ -483,6 +483,54 @@ public void decryptData(String templateName) throws CryptoException {
     encryptOrDecryptData(templateName, false);
   }
 
+  public List<DataRecord> getRecords(IdentifiedRecordTemplate template, String fieldName,
+      String fieldValue) throws FormException {
+    Connection con = null;
+    try {
+      con = getConnection();
+      return selectRecords(con, template, fieldName, fieldValue);
+    } catch (SQLException e) {
+      throw new FormException("GenericRecordSetManager.getRecords", "form.EXP_SELECT_FAILED", e);
+    } finally {
+      closeConnection(con);
+    }
+  }
+
+  private List<DataRecord> selectRecords(Connection con, IdentifiedRecordTemplate template,
+      String fieldName, String fieldValue) throws SQLException, FormException {
+    List<DataRecord> records = new ArrayList<>();
+    PreparedStatement select = null;
+    ResultSet rs = null;
+
+    try {
+      select = con.prepareStatement(SELECT_TEMPLATE_RECORDS_BY_FIELDVALUE);
+
+      select.setInt(1, template.getInternalId());
+      select.setString(2, fieldName);
+      select.setString(3, fieldValue);
+      select.setString(4, "%##"+fieldValue);
+      select.setString(5, "%##"+fieldValue+"##%");
+      select.setString(6, fieldValue+"##%");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 6 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)